### PR TITLE
fix(board): add board write tool timeout

### PR DESCRIPTION
**Pull request by jeong-sik**
## Summary

- add `MASC_TOOL_TIMEOUT_BOARD_SEC` for board write MCP tools, defaulting to 90s and clamped to 5s..300s
- keep regular bounded MCP tools on `MASC_TOOL_TIMEOUT_DEFAULT_SEC` with the existing 60s default
- document the tools/call timeout layer and add focused tests for board default, override, and clamp behavior

## Review Follow-up

- Expanded the board write timeout predicate to include board admin mutations: `keeper_board_delete`, `keeper_board_cleanup`, `masc_board_delete`, and `masc_board_cleanup`.
- Updated the interface comment so operators can see that post/comment/vote/comment_vote/curation_submit/delete/cleanup and `masc_board_reaction` use the board timeout bucket.
- Extended the timeout test matrix so admin board mutations are covered by the board-specific timeout default.

## Why

#10569 showed board write RPCs (`keeper_board_post`, comments, votes) all hitting the generic 60s tools/call cap while board persistence can queue behind the JSONL persist mutex. #13038 already added persist-lock histograms; this PR adds a bounded board-specific override so operators can mitigate queue pressure without changing synchronous durability semantics.

This is a mitigation, not the full async board-writer redesign.

## Validation

- `git diff --check origin/main..HEAD`
- `MASC_SKIP_OPAM_LOCK=1 MASC_SKIP_PIN_CHECK=1 MASC_DUNE_THROTTLE=0 DUNE_BUILD_DIR=_build_codex_13678_review scripts/dune-local.sh exec ./test/test_mcp_server_eio_call_tool.exe -- test timeout` was attempted but the Alcotest filter matched no tests.
- `./_build_codex_13678_review/default/test/test_mcp_server_eio_call_tool.exe` -> 12 tests

Refs #10569

### CODE REVIEW

**Review by anyang-keepers — COMMENTED** [review]
## qa-king — board write timeout: 4 boundary tests missing

**Positive**: 59 lines of test code, 12 tests passing, env var override documented, clamp behavior tested. `git diff --check` clean. Copilot 2 inline comments.

### QA Gap 1: Clamp boundary off-by-one

Tests cover default, override, and clamp concept — but not the exact boundary values:

```ocaml
(* MISSING *)
let test_clamp_boundaries () =
  (* MASC_TOOL_TIMEOUT_BOARD_SEC=4 → clamped to 5 *)
  (* MASC_TOOL_TIMEOUT_BOARD_SEC=5 → accepted as-is *)
  (* MASC_TOOL_TIMEOUT_BOARD_SEC=300 → accepted as-is *)
  (* MASC_TOOL_TIMEOUT_BOARD_SEC=301 → clamped to 300 *)
  (* MASC_TOOL_TIMEOUT_BOARD_SEC=0 → what happens? 5 or crash? *)
  (* MASC_TOOL_TIMEOUT_BOARD_SEC=-1 → what happens? *)
  ()
```

If `Int.of_string` on env returns 0 or negative, is that handled? The clamp is `max 5 (min 300 value)` — if `value < 0`, `max 5 -1 = 5`? Verify.

### QA Gap 2: Board tool vs non-board tool classification — no negative test

PR body says board write tools get 90s, others stay at 60s. But what about **board READ tools**?

```ocaml
(* MISSING *)
let test_board_read_uses_default_not_board_timeout () =
  (* keeper_board_list, keeper_board_get → should use MASC_TOOL_TIMEOUT_DEFAULT_SEC *)
  (* NOT MASC_TOOL_TIMEOUT_BOARD_SEC *)
  (* If board reads use the board timeout, that's a bug *)
  ()
```

The diff adds `is_board_write_tool` classification — does it accidentally include read tools?

### QA Gap 3: Timeout expiry behavior — kill vs. grace

When timeout fires, what happens to the in-flight board write?

```ocaml
(* MISSING *)
let test_timeout_expiry_does_not_corrupt_board_state () =
  (* Start board post that takes 100s (simulate slow persist) *)
  (* Timeout fires at 90s *)
  (* Assert: board file is NOT half-written *)
  (* Assert: board file is NOT missing the entry *)
  (* Assert: board file is consistent (all entries valid JSONL) *)
  ()
```

If timeout kills the fiber mid-JSONL-write, the board file could be corrupted. Atomic write (write-to-temp + rename) would prevent this, but is that documented/tested?

### QA Gap 4: Mutex contention under timeout — deadlock risk

PR body says "board persistence can queue behind the JSONL persist mutex." If a board write times out while holding the mutex, does the mutex release?

```ocaml
(* MISSING *)
let test_mutex_released_on_timeout () =
  (* Thread 1: board write acquires persist mutex, timeout fires *)
  (* Thread 2: board write attempts to acquire persist mutex *)
  (* Assert: Thread 2 does NOT deadlock *)
  (* Assert: Thread 2 completes within reasonable time *)
  ()
```

Without this, timeout "solves" the user-facing hang but creates a mutex leak that eventually blocks all board writes.

### Verdict

| 항목 | 상태 | merge blocker? |
|---|---|---|
| Clamp boundary (4/5/300/301/0/-1) | ❌ | **YES** |
| Board read vs write classification | ❌ | **YES** |
| Timeout + board file integrity | ❌ | 권장 |
| Mutex release on timeout | ❌ | **YES** (deadlock risk) |

**3 merge blockers.** Mutex leak under timeout is the most dangerous — it converts a transient hang into a permanent deadlock. Clamp boundaries are table stakes. Board read classification is a correctness issue.

File: test/test_mcp_server_eio_call_tool.ml [hunk 1]
```diff
@@ -166,27 +166,56 @@ let test_regular_tool_uses_default_timeout () =
...
+      "masc_board_post";
+      "masc_board_comment";
+      "masc_board_vote";
+      "masc_board_comment_vote";
+      "masc_board_delete";
+      "masc_board_cleanup";
+      "masc_board_reaction";
+      "masc_board_curation_submit";
+      "masc_board_delete";
+      "masc_board_cleanup";
```
[the remark this answers is not in the document]
Comment (jeong-sik) [reply test/test_mcp_server_eio_call_tool.ml]: Resolved on current head 5dba2b2545: the board write timeout test matrix now lists each board write tool once, including masc_board_delete and masc_board_cleanup only once.
